spec/controller_services/shopping_list_items_controller/update_service_spec.rb

```diff
@@ -0,0 +1,148 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require 'service/ok_result'
+require 'service/not_found_result'
+require 'service/method_not_allowed_result'
+require 'service/unprocessable_entity_result'
+
+RSpec.describe ShoppingListItemsController::UpdateService do
+  describe '#perform' do
+    subject(:perform) { described_class.new(user, list_item.id, params).perform }
+
+    let(:user) { create(:user) }
+    let(:shopping_list) { user.shopping_lists.create! }
+    
+    context 'when all goes well' do
+      let!(:list_item) { create(:shopping_list_item, list: shopping_list, quantity: 2) }
+      let(:params) { { quantity: 3 } }
+      let(:master_list) { user.master_shopping_list }
+      let(:scope) { ShoppingListItem.belonging_to_user(user) }
+
+      before do
+        master_list.add_item_from_child_list(list_item)
+      end
+
+      it 'updates the shopping list item', :aggregate_failures do
+        perform
+        expect(list_item.reload.quantity).to eq 3
+        expect(list_item.reload.notes).to be nil
+      end
+
+      it 'updates the item on the master list' do
+        # Put the mocks in here because I don't want this much mocking in the other specs
+        allow(ShoppingListItem).to receive(:belonging_to_user).with(user).and_return(scope)
+        allow(scope).to receive(:find).and_return(list_item)
+        allow(list_item).to receive(:list).and_return(shopping_list)
+        allow(shopping_list).to receive(:master_list).and_return(master_list)
+        allow(master_list).to receive(:update_item_from_child_list)
+        perform
+        expect(master_list).to have_received(:update_item_from_child_list).with(list_item.description, 1, nil, nil)
+      end
+
+      it 'returns a Service::OKResult' do
+        expect(perform).to be_a(Service::OKResult)
+      end
+
+      it 'sets the resource to include both the regular and master list item' do
+        expect(perform.resource).to eq([master_list.list_items.first, list_item])
+      end
+
+      it 'updates the updated_at timestamp on the list' do
+        Timecop.freeze do
+          perform
+          # This is another case of a rounding error in the CI environment. The server where
+          # GitHub Actions run seems to truncate the last few digits of the timestamp, resulting
+          # in things being not quite equal in that environment. Since it's not that important
+          # that it be that exact, I'm just using the `be_within` matcher.
+          expect(shopping_list.reload.updated_at).to be_within(0.05.seconds).of(Time.now.utc)
```

Comment:
I could probably tighten the tolerance further since it's only the last few digits, but 0.05 is honestly good enough. Users are unlikely to be able to make requests from the UI fast enough for race conditions to come up as a result of the timestamps not being perfect.